lib/ledenet/api.rb

```diff
@@ -14,56 +14,112 @@ def initialize(device_address, options = {})
 
     def on
       send_bytes_action(0x71, 0x23, 0x0F, 0xA3)
-
       true
     end
 
     def off
       send_bytes_action(0x71, 0x24 ,0x0F, 0xA4)
-
       true
     end
 
     def on?
-      (status.bytes[13] & 0x01) == 1
+      # Bulbs respond with "35" or "36" (HEX: 23 or 24)
+      # for on and off respectively
+      power_state = status[2].unpack('C').to_s.delete('[]')
+      if  power_state == "35"
+        true
+      else
+        false
+      end
     end
 
-    def update_color(r, g, b)
-      checksum = color_checksum(r, g, b)
+    def update_color(r,g,b) # Supports legacy color Updates
+      update_ufo(r,g,b,0,false)
+      true
+    end
 
-      send_bytes_action(0x31, r, g, b, 0xFF, 0, 0x0F, checksum)
+    def update_ufo(r,g,b,w,persist) # Update a UFO wireless device
+      msg = Array.new
+      if persist
+        msg << 0x31
+      else
+        msg << 0x41 << r << g << b << w << 0x00 << 0x0f
+      end
+      checksum = calc_checksum(msg)
+      send_bytes_action(*msg, checksum)
+      true
+    end
 
+    def update_bulb_color(r, g, b, persist) # Update a Bulb wireless device's color
+      msg = Array.new
+      if persist
+        msg << 0x31
+      else
+        msg << 0x41 << r << g << b << 0x00 << 0xf0 << 0x0f
+      end
+      checksum = calc_checksum(msg)
+      send_bytes_action(*msg, checksum)
       true
     end
 
-    def current_color
-      status.bytes[6, 3]
+    def update_bulb_white(w, persist) # Update a Bulb wireless device's WW level
+    msg = Array.new
+    if persist
+        msg << 0x31
+      else
+        msg << 0x41 << 0x00 << 0x00 << 0x00 << w << 0x0f << 0x0f
+      end
+        checksum = calc_checksum(msg)
+        send_bytes_action(*msg, checksum)
+        true
+    end
+
+    def current_status # Gets bytes 6-9 and returns them as Integers from 0-255 (Red,Green,Blue, and WW) and return power status as string
+      current_packet = Array.new
+      current_packet = status
+      power_state = "off"
+      power_state = "on" if current_packet[2].unpack('C').to_s.delete('[]') == "35"
+      return Integer(current_packet[6].unpack('C').to_s.delete('[]')).to_i,Integer(current_packet[7].unpack('C').to_s.delete('[]')).to_i,Integer(current_packet[8].unpack('C').to_s.delete('[]')).to_i,Integer(current_packet[9].unpack('C').to_s.delete('[]')).to_i, power_state
     end
 
+    def current_color
+      current_packet = Array.new
+      current_packet = status
+      currentColors = Array.new
```

Comment:
Think it'd be cleaner to just set the array instead of initializing and then appending to it. Also would prefer to use snake casing consistently (`current_colors` instead of `currentColors`).